src/analysis/retail/shaman/enhancement/modules/talents/HotHand.tsx

```diff
@@ -207,19 +208,41 @@ class HotHand extends MajorCooldown<HotHandProc> {
   }
 
   startOrRefreshWindow(event: ApplyBuffEvent | RefreshBuffEvent) {
-    // on application both resets the CD and applies a mod rate
-    this.spellUsable.endCooldown(TALENTS.LAVA_LASH_TALENT.id, event.timestamp);
+    // on application applies a mod rate and also resets CD if proc was natural
+
+    const whirlingFireRemovedEvent = GetRelatedEvent<RemoveBuffEvent>(
+      event,
+      EnhancementEventLinks.WHIRLING_FIRE_LINK,
+      (e) => e.type === EventType.RemoveBuff,
+    );
+
+    // cooldown isn't reset if the Hot Hands stems from Whirling Fire
+    if (!whirlingFireRemovedEvent) {
+      this.spellUsable.endCooldown(TALENTS.LAVA_LASH_TALENT.id, event.timestamp);
+    }
 
     if (!this.activeWindow) {
       this.spellUsable.applyCooldownRateChange(TALENTS.LAVA_LASH_TALENT.id, this.hotHand.rate);
       this.hotHandActive.startInterval(event.timestamp);
 
+      // make sure to include first Lava Lash of the window, when triggered by Whirling Fire
+      let lavaLashCastEvent: CastEvent | undefined;
+      if (whirlingFireRemovedEvent) {
+        lavaLashCastEvent = GetRelatedEvent<CastEvent>(
+          event,
+          EnhancementEventLinks.WHIRLING_FIRE_LINK,
+          (e) => e.type === EventType.Cast,
+        );
+      }
```

Comment:
We can add additional information to the timeline to indicate this Hot Hand was triggered by Whirling Fire. E.g.
```suggestion
      let lavaLashCastEvent: CastEvent | undefined;
      if (whirlingFireRemovedEvent) {
        lavaLashCastEvent = GetRelatedEvent<CastEvent>(
          event,
          EnhancementEventLinks.WHIRLING_FIRE_LINK,
          (e) => e.type === EventType.Cast,
        );
        lavaLashCastEvent && addAdditionalCastInformation(
          lavaLashCastEvent,
          <>
            <SpellLink spell={TALENTS.HOT_HAND_TALENT} /> was applied by{' '}
            <SpellLink spell={SPELLS.WHIRLING_FIRE} />
          </>,
        );
      }
```
Or `addEnhancedCastReason` if you think it's more appropriate.